File: cpp/405_Convert_a_Number_to_Hexadecimal.py
class Solution:
    def toHex(self, num: int) -> str:
        if num == 0:
            return "0"

        n = num & 0xFFFFFFFF

        hex_chars = "0123456789abcdef";
        ret = "";

        while n:
            ret += hex_chars[n & 0xF]
            n >>= 4

        return ret[::-1]

File: cpp/405_Convert_a_Number_to_Hexadecimal.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    string toHex(int num) {
        if (num == 0)
            return "0";

        unsigned int n = num;

        string hex = "0123456789abcdef";
        string ret = "";

        while (n) {
            ret += hex[n & 0xF];
            n >>= 4;
        }

        reverse(ret.begin(), ret.end());

        return ret;
    }
};
